Automattic/pocket-casts-android
# Add audio clip sharing

## Description

This PR adds audio clip sharing. For now, the clip is always shared as a sheet due to the demo UI.

Clip creation is done with `FFmpeg` by combining two streams. The first stream is an audio stream, which comes either from a downloaded file or from a URL. The second, optional stream, is for embedding covers into mp3 files. We cannot rely on the cover embedded in the original file because, firstly, it might not exist, and secondly, the cover is always represented as a video stream. For video files, it might be actual video instead of a cover. Downloading the cover also means it won't be present if someone creates a clip from a downloaded file while being offline. This is an understandable compromise given the limitations of the whole flow. Then, FFmpeg re-encodes the audio stream to an mp3 file. I decided not to use the original codec and re-encode to mp3 for consistency.

Another issue noticed is that Tumblr and X do not support sharing `audio/mp3`, so this will need to be addressed in the designs. Neither does Instagram directly but they have a separate API for that which we will use once implemented.

I wanted to write automatic tests for `FFMpegMediaService`, but with the current Gradle and CI setup, it is not possible.

## Testing Instructions

> [!note]
> Audio clip sharing for now works only with `debug` and `debugProd` variants.

1. Go to the clip creation.
2. Share a clip as audio.
3. Wait until clip creation finishes and share it somewhere like Slack.
4. Play the clip and check if it matches. Note that it might not be 100% accurate. This is something I plan to investigate: p1721029505573319/1721021230.484059-slack-C028JAG44VD
5. Download the clip and check if it has a cover.

Run this scenario for different podcasts, a video podcast, and a downloaded file.

## Checklist
